core-ui/src/components/Predefined/Create/Roles/helpers.js

```diff
@@ -1,3 +1,21 @@
+export function unique(arr) {
+  return [...new Set(arr)];
+}
+export const EMPTY_API_GROUP_KEY = 'core-api-group';
+export const extractApiGroup = groupVersion => {
+  // handle core ('') group
+  if (groupVersion === 'v1') return '';
+  const [apiGroup] = groupVersion.split('/');
+  return apiGroup;
+};
+
+export const getApiGroupInputOptions = groupVersions =>
+  unique(groupVersions.map(extractApiGroup)).map(g =>
```

Comment:
can we please add either `?` before map, or `|| []` at the end? I think it's the most common cause when it comes to crashing UIs in our code 😄 

(please also apply in other places, it doesn't make sense to comment it all - also for 'some', 'filter' etc)